extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -183,7 +217,111 @@ export class ResourceTypePickerDialog extends DialogBase {
 		this._dialogObject.content = [tab];
 	}
 
+	private createResouceButtonsContainer(): azdata.ListViewComponent {
+		const tags = this.getAllResourceTags(this._resourceTypes);
+		if (!tags.includes('All')) {
+			tags.splice(0, 0, 'All');
+		}
+		const items: azdata.ListViewOption[] = [];
+		tags.forEach((t: string) => {
+			items.push({
+				label: t,
+				id: t
+			});
+		});
+		const listView = this._view.modelBuilder.listView().withProps({
+			title: {
+				text: 'Categories',
+			},
+			CSSStyles: {
+				'width': '140px'
+			},
+			options: items,
+			selectedOptionId: items[0].id
+		}).component();
+		this._toDispose.push(listView.onDidClick((e) => {
+			this.filterResources();
+		}));
+
+		return listView;
+	}
+
+	private filterResources(): void {
+
+		let search = this._resourceSearchBox.value;
```

Comment:
nit: Simpler as `const search = this._resourceSearchBox.value?.toLowerCase() ?? ''`;